plugins/outputs/sql/sql.go

```diff
@@ -248,15 +251,36 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 		}
 
 		sql := p.generateInsert(tablename, columns)
-		_, err := p.db.Exec(sql, values...)
 
-		if err != nil {
-			// check if insert error was caused by column mismatch
-			p.Log.Errorf("Error during insert: %v, %v", err, sql)
-			return err
+		switch p.Driver {
+		case "clickhouse":
+			// ClickHouse needs to batch inserts with prepared statements
+			tx, err := p.db.Begin()
+			if err != nil {
+				return fmt.Errorf(": %v", err)
```

Comment:
There is something missing I guess... Maybe
```suggestion
				return fmt.Errorf("begin failed: %v", err)
```